### PR TITLE
Issue #73 and update pom.xml

### PR DESCRIPTION
**Pull request by ReeceSX**
Addresses an issue with some synthetic functions 
https://github.com/md-5/SpecialSource/issues/73

### CODE REVIEW

File: src/main/java/net/md_5/specialsource/JarMapping.java [hunk 1]
```diff
@@ -608,12 +609,14 @@ private void parseSrgLine(String line, MappingTransformer inputTransformer, Mapp
...
             String oldEntry = oldClassName + "/" + oldFieldName;
+            String newEntryHack = newClassName + "/" + oldFieldName;
+
             if (fields.containsKey(oldEntry) && !newFieldName.equals(fields.get(oldEntry))) {
                 throw new IllegalArgumentException("Duplicate field mapping: " + oldEntry + " ->" + newFieldName
                         + " but already mapped to " + fields.get(oldEntry) + " in line=" + line);
             }
-
             fields.put(oldEntry, newFieldName);
+            fields.put(newEntryHack, newFieldName);
```
Comment (md-5) [inline src/main/java/net/md_5/specialsource/JarMapping.java:619]: I don't understand why the changes in this class are needed. They would break mapping where you have a situation like:

a -> b
b -> a

Shouldn't visitInvokeDynamicInsn be getting the old information anyway, why does it need the new one?

Comment (ReeceSX) [reply src/main/java/net/md_5/specialsource/JarMapping.java:619]: I'm aware. As I said I don't know how to best approach this in the issue. By the time the dynamic invoker visitor is called there are seemingly no references to the old symbol with the parent already being renammed. I don't think this is an issue for Minecraft, at least. I was hoping you could offer a suggestion on how to pull the old class name from the new mapping.

Comment (md-5) [reply src/main/java/net/md_5/specialsource/JarMapping.java:619]: I'm not sure why that would be the case, you might want to step through the call to visitInvokeDynamic with a debugger to see where it is getting the new symbol.
SS supports more than just Minecraft, and this would be an issue even for some Minecraft mapping sets used.

Comment (ReeceSX) [reply src/main/java/net/md_5/specialsource/JarMapping.java:619]: Hacking in an alternative hashtable and one `public String mapFieldNameAlternative(String owner, String name, String desc, int access)` stub under CustomRemapper

Comment (ReeceSX) [reply src/main/java/net/md_5/specialsource/JarMapping.java:619]: > I'm not sure why that would be the case, you might want to step through the call to visitInvokeDynamic with a debugger to see where it is getting the new symbol.
> SS supports more than just Minecraft, and this would be an issue even for some Minecraft mapping sets used.

Dammit, you're right, this isn't the case. This fix was only relevant to release < SpecialSource-1.9.0.  The bsmArgs are indeed different

Comment (ReeceSX) [reply src/main/java/net/md_5/specialsource/JarMapping.java]: Reverted the now irrelevant and problematic changes. Confirmed it looks good under Decaf. Relinking my test client now as as an acceptance test.

**Comment by md-5** [comment]
Thanks
